xdsl/dialects/memref.py

```diff
@@ -472,6 +474,43 @@ def get(
         return Cast.build(operands=[source], result_types=[type])
 
 
+@irdl_op_definition
+class DmaWaitOp(IRDLOperation):
+    name = "memref.dma_wait"
+
+    tag: Annotated[Operand, MemRefType]
+    tag_indices: Annotated[VarOperand, IndexType]
+
+    num_elements: Annotated[Operand, IndexType]
+
+    @staticmethod
+    def get(
+        tag: SSAValue | Operation,
+        tag_indices: Sequence[SSAValue | Operation],
+        num_elements: SSAValue | Operation,
+    ):
+        return DmaWaitOp.build(
+            operands=[
+                tag,
+                tag_indices,
+                num_elements,
+            ]
+        )
+
+    def verify_(self) -> None:
+        assert isa(self.tag.typ, MemRefType[Attribute])
+
+        if len(self.tag.typ.shape) != len(self.tag_indices):
+            raise VerifyException(
+                "Expected {} tag indices because of shape of tag memref".format(
```

Comment:
nit: 
```suggestion
                f"Expected {len(self.tag.typ.shape)} tag indices because of shape of tag memref".format(
```

Reply:
fixed in 87fa7d28246e7ef757fad9e50091b56e927be369